### PR TITLE
Allow to setup LDAP auth

### PR DESCRIPTION
**Pull request by mescanef**
Be able to setup LDAP auth on Jenkins master. This resulted in splitting
set security method into three methods for setting:

ldap auth
password auth
unsecured
Also give choice to turn on the slave to master feature during
deployment.

### CODE REVIEW

File: manifests/security.pp [hunk 1]
```diff
@@ -17,17 +17,91 @@
...
 class jenkins::security (
-  $security_model = undef,
+  $jenkins_management_email         = '',
+  $jenkins_management_login         = '',
+  $jenkins_management_name          = '',
+  $jenkins_management_password      = '',
+  $jenkins_s2m_acl                  = false,
+  $jenkins_ssh_private_key_contents = '',
+  $jenkins_ssh_public_key_contents  = '',
+  $ldap_access_group                = '',
```
Comment (rtyler) [inline manifests/security.pp]: This is a huge amount of parameters for the jenkins::security class. I suggest changing this to take a hash called `ldap_config`. You could even set the defaults in a container class like `jenkins::ldap::config` or something like that.

File: manifests/security.pp [hunk 2]
```diff
@@ -17,17 +17,91 @@
...
+  $ldap_user_search_base            = '',
+  $security_model                   = undef,
 ){
   validate_string($security_model)
 
   include ::jenkins::cli_helper
 
+  $libdir = $::jenkins::libdir
+
+  file { "${libdir}/.ssh/id_rsa" :
```
Comment (rtyler) [inline manifests/security.pp]: The management of some Jenkins public/private SSH keys seems unrelated to this LDAP configuration change, what's it here for?

File: manifests/security.pp [hunk 3]
```diff
@@ -17,17 +17,91 @@
...
+    content => "${jenkins_ssh_public_key_contents} jenkins@${::fqdn}",
+    replace => true,
+    require => File["${libdir}/.ssh"],
+  }
+
+  if ($security_model == 'unsecured') {
+    $security_opt_params = 'set_security_unsecured'
+  }
+
+  if ($security_model == 'ldap') {
```
Comment (rtyler) [inline manifests/security.pp]: One suggestion, since this is a lot to place on the command line. What about writing a configuration file to disk temporarily and then having the groovy helper read that configuration file?

I worry that there command line might end up being too long

Comment (jhoblitt) [inline manifests/security.pp]: In my native types and provider branch I've been passing json into the puppet_helper on the stdin, which has worked well.

Comment (jhoblitt) [inline manifests/security.pp]: Here's a specific example of handling json input: https://github.com/jhoblitt/puppet-jenkins/blob/feature/types_and_providers/files/puppet_helper.groovy#L209-L212

**Comment by jhoblitt** [comment]
@mescanef My work to support configuring the security realm and authentication strategy via native types was merged in #382.  Do you think this work could be merge with the groovy `get_security_realm()` / `get_authorization_strategy()` methods on master and the providers for `jenkins_security_realm` and `jenkins_authorization_strategy`?  @daenney is also interested in configuring ldap.

**Comment by rhoml** [comment]
Any news on this PR?

**Comment by jhoblitt** [comment]
@rhoml At a minimum it needs to be rebased on the current master and I think it should be converted to passing json instead of CLI args.  I think it would be less work to add this to the `jenkins_security_realm/jenkins_authorization_strategy` types.

File: manifests/security.pp (hunk 3 above)
Comment (matonb) [inline manifests/security.pp]: Nothing like reviving old branches...

@jhoblitt Your example json URL appears to be dead, are you still using this branch to setup Jenkins LDAP authentication ?

Comment (jhoblitt) [reply manifests/security.pp]: @matonb That code has been merged into this module.

**Comment by bastelfreak** [comment]
Hello everybody. I'm going to close this PR due to inactivity. Please reopen and rebase the PR if anybody is still interested in it.
